### PR TITLE
[11.0][MIG] product_supplierinfo_for_customer_sale

### PR DESCRIPTION
**Pull request by xavierjimenez**
Migration of the module **product_supplierinfo_for_customer_sale** to version 11.0.

Depends on:

- [x] https://github.com/OCA/product-attribute/pull/367 product_supplierinfo_for_customer

### CODE REVIEW

**Review by HaraldPanten — APPROVED** [review]
Functionally working fine :+1:

File: product_supplierinfo_for_customer_sale/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2013-2017 Agile Business Group sagl
+#     (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Product Supplierinfo for Customer Sale",
+    "version": "11.0.1.0.0",
+    "summary": "Loads in every sale order line the customer code defined "
+               "in the product",
+    "author": "Agile Business Group,Odoo Community Association (OCA)",
+    "website": "http://www.agilebg.com",
```
Comment (nikul-serpentcs) [inline product_supplierinfo_for_customer_sale/__manifest__.py]: @xavierjimenez IMHO Use URL as per https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10

File: product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        """Create a Partner."""
+        return self.env['res.partner'].create({
+            'name': name,
+            'email': 'example@yourcompany.com',
+            'customer': True,
+            'phone': 123456,
+        })
+
+    def _create_supplierinfo(self, type, partner, product):
+        return self.env['product.supplierinfo'].create({
```
Comment (nikul-serpentcs) [inline product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py]: @xavierjimenez IMHO use `supplierinfo_model`

File: product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            'product_id': product.id,
+            'product_name': 'product4',
+            'product_code': '00001',
+            'type': type,
+            'price': 100.0,
+            'min_qty': 15.0,
+        })
+
+    def _create_pricelist(self, name, product):
+        return self.env['product.pricelist'].create({
```
Comment (nikul-serpentcs) [inline product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py]: @xavierjimenez IMHO directly use `pricelist_model`.

File: product_supplierinfo_for_customer_sale/__manifest__.py (hunk 1 above)
Comment (moylop260) [reply product_supplierinfo_for_customer_sale/__manifest__.py]: In fact, the [template module](https://github.com/OCA/maintainer-tools/blob/9766529146bcb61ce746623da74a23163199bdd3/template/module/__manifest__.py#L11) has https://github.com/OCA/sale-workflow/blob/11.0/product_supplierinfo_for_customer_sale

File: product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py (hunk 2 above)
Comment (moylop260) [reply product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py]: You mean, `self.supplierinfo_model`
+1

**Comment by moylop260** [comment]
@xavierjimenez 

I just fixed the oca_dependencies to download the dependency module.

Using runbot and testing the feature I saw the following error:
 - sale_order_line

```bash
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/fields.py", line 937, in __get__
    value = record.env.cache.get(record, self)
  File "/.repo_requirements/odoo/odoo/api.py", line 960, in get
    value = self._data[field][record.id][key]
KeyError: <odoo.api.Environment object at 0x7f8d244a4320>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 936, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 934, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/OCB-11.0/addons/web/controllers/main.py", line 926, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/models.py", line 5103, in onchange
    record.mapped(dotname)
  File "/.repo_requirements/odoo/odoo/models.py", line 4512, in mapped
    recs = recs._mapped_func(operator.itemgetter(name))
  File "/.repo_requirements/odoo/odoo/models.py", line 4491, in _mapped_func
    vals = [func(rec) for rec in self]
  File "/.repo_requirements/odoo/odoo/models.py", line 4491, in <listcomp>
    vals = [func(rec) for rec in self]
  File "/.repo_requirements/odoo/odoo/models.py", line 4758, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/.repo_requirements/odoo/odoo/fields.py", line 943, in __get__
    self.determine_draft_value(record)
  File "/.repo_requirements/odoo/odoo/fields.py", line 1063, in determine_draft_value
    self._compute_value(record)
  File "/.repo_requirements/odoo/odoo/fields.py", line 999, in _compute_value
    getattr(records, self.compute)()
  File "/home/odoo/build/OCA/sale-workflow/product_supplierinfo_for_customer_sale/models/sale.py", line 22, in _compute_get_product_customer_code
    ], limit=1)
  File "/home/odoo/dependencies/product-attribute/product_supplierinfo_for_customer/models/product_supplierinfo.py", line 35, in search
    args, offset=offset, limit=limit, order=order, count=count)
  File "/.repo_requirements/odoo/odoo/models.py", line 1480, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/.repo_requirements/odoo/odoo/models.py", line 3773, in _search
    query = self._where_calc(args)
  File "/.repo_requirements/odoo/odoo/models.py", line 3563, in _where_calc
    e = expression.expression(domain, self)
  File "/.repo_requirements/odoo/odoo/osv/expression.py", line 668, in __init__
    self.parse()
  File "/.repo_requirements/odoo/odoo/osv/expression.py", line 846, in parse
    raise ValueError("Invalid field %r in leaf %r" % (left, str(leaf)))
ValueError: Invalid field 'type' in leaf "<osv.ExtendedLeaf: ('type', '=', 'customer') on product_supplierinfo (ctx: )>"
```

**Comment by moylop260** [comment]
Another comment is that I searched from sale.order.line using the customer product code but is not found it the product like the old behaviour:
 - https://github.com/OCA/product-attribute/blob/87e9ca857e4ac8ccbcb7fc6c3792c5f796b72b88/product_supplierinfo_for_customer_sale/models/product_product.py#L13-L34

File: product_supplierinfo_for_customer_sale/__init__.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Copyright 2013-2017 Agile Business Group sagl
```
Comment (nikul-serpentcs) [inline product_supplierinfo_for_customer_sale/__init__.py]: @xavierjimenez  Remove Copyright from all `__init__` file

**Review by nikul-serpentcs — COMMENTED** [review]
Minor Change

**Comment by moylop260** [comment]
Hi @xavierjimenez 
I’m interesting to migrate this module.
Tell me if you are working on this or prefer that I open another PR

**Comment by xavierjimenez** [comment]
Hi @moylop260,

I'm sorry but I'm pretty busy right now so I don't think I can finish it soon. You can open a new PR if you want and I'll close this one, thanks.

**Comment by moylop260** [comment]
I just changed the name of the field from `type` to `supplierinfo_type` and now all is working.

 - sale

**Comment by moylop260** [comment]
I just get `name_search` functionality and commits from:
 - https://github.com/OCA/product-attribute/tree/8.0/product_supplierinfo_for_customer_sale

Including tests, readme, authors and so on.

And now is working the searching too:
 - 
sale2

**Review by moylop260 — APPROVED** [review]

**Comment by moylop260** [comment]
@pedrobaeza 
Could you check it, please?

**Comment by JulioSerna** [comment]
LGTM 👍

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional test passed

**Comment by rafaelbn** [comment]
Please, take a look in : https://github.com/OCA/sale-workflow/issues/737

**Pull request merged**